### ISSUE TITLE
Integration with intellij-elm?

### ISSUE THREAD

**Issue by klazuka** [frawa/intellij-elm-test-runner#8]
@frawa I was looking into building elm-test support for intellij-elm when I discovered your project. I wish I would have known about it earlier--I've been running my tests from the command-line all this time!

Do you have any objections to me folding this functionality into the main plugin? Is there a reason why it should be kept separate? Is there anything that you would have done differently if you were to do it again?

Thanks 🙏

**Comment by frawa** [comment]
@klazuka Merging this plugin into yours makes totally sense. I consider the current functionality stable enough to make the move. (I am using it daily without the desire to change since weeks now.)

I still have many ideas for additional features, but those can be added later.
The current approach seems reasonable to keep, it is my already my second attempt, see https://github.com/frawa/vscode-elm-test-runner.

Let me know how I can help merging ...

**Comment by klazuka** [comment]
I just read all of your code and mapped it out, so I have a pretty good understanding now. IntelliJ's boiler plate around run configurations is annoying, but your code was simple and easy to understand 👍 

I suspect that the Java code could just be copied over, and the 2 extension points added to `plugin.xml`. And we would definitely want to copy over the tests.

The gson dependency would have to be removed in favor of Jackson.

The configuration of the Elm folder (for the working directory) could be changed, perhaps, to list the attached Elm projects in the `ElmWorkspaceService` and have the user choose one.

The path to `elm-test` binary should probably be moved into `ElmToolchain` and the settings stored as part of `ElmWorkspaceService`. Currently we just store the path to the directory that contains the `elm` binary. The hope was that things like `elm-format` and `elm-test` would always be located in the same directory. But I don't think that's always true (depending on which OS and installer they use), and we probably will need to start storing separate paths to each directory.

And eventually I would want to port the code to Kotlin.

This is just my rough outline of the tasks involved. If you would like to do the work, that'd be fine with me, and I'd be happy to do the code review and provide additional guidance. Otherwise, I'll do the work. Just let me know.

**Comment by FloWi** [comment]
@klazuka Have you tried IDEA's paste-java-code-into-{otherLanguage}-file feature?

![bildschirmfoto 2019-02-19 um 19 13 37](https://user-images.githubusercontent.com/39078/53037474-9c08fd00-347a-11e9-856c-b3ee0814552e.png)

For scala it's a bit brittle, but it looks like it works quite well for kotlin. No wonder since the Jetbrains pro's wrote the kotlin plugin :) 

Maybe you give it a go and see if the result works for you. I suggest you start with classes that don't have a lot dependencies and move your way up the dependency tree. This way it's easier with the imports.

**Comment by klazuka** [comment]
Yeah, I've used it before and it works pretty well. The generated Kotlin code still needs another pass over it to make it more idiomatic, but it certainly makes the job easier.

However, in this case, I think it makes sense to first integrate the Java code as-is. Get the tests passing with as few code changes as possible. And then port to Kotlin.

**Comment by FloWi** [comment]
Sounds reasonable. Just wanted to make sure you know this feature and don't type too much :)

**Comment by frawa** [comment]
For reference https://github.com/klazuka/intellij-elm/pull/301
